keras/callbacks.py

```diff
@@ -214,17 +227,23 @@ def on_batch_end(self, batch, logs=None):
         self.seen += batch_size
 
         for k, v in logs.items():
-            if k in self.totals:
-                self.totals[k] += v * batch_size
+            if k in self.stateful_metrics:
+                self.totals[k] = v
             else:
-                self.totals[k] = v * batch_size
+                if k in self.totals:
+                    self.totals[k] += v * batch_size
+                else:
+                    self.totals[k] = v * batch_size
 
     def on_epoch_end(self, epoch, logs=None):
         if logs is not None:
             for k in self.params['metrics']:
                 if k in self.totals:
                     # Make value available to next callbacks.
```

Comment:
"make value available to future callbacks"

Reply:
Callbacks are processed sequentially, future is more vague than next IMO.

Reply:
I don't think the grammar is right as-is with "next".

Alternate: "make value available to each of the following callbacks"